### ISSUE TITLE
Re-implement Environment

### ISSUE THREAD

**Issue by okainov** [kiwitcms/Kiwi#1344]
### Description of problem

Tags are not flexible enough to handle complex Test Run environments (version of dependencies, OS, type of test run and many others). This is a blocking issue for our team and prevents us from further upgrades of Kiwi.
We rely on different properties and organize our test reports accordingly.

Example of our typical test run:
![image](https://user-images.githubusercontent.com/918446/73856739-369e4f80-4836-11ea-81fb-cf5a41b5492b.png)

Please add possibility to specify organized Environment properties and values back.

### Version or commit hash (if applicable)

Any after https://github.com/kiwitcms/Kiwi/pull/485

**Comment by aglassman** [comment]
I was looking forward to using this feature after looking through the docs, but couldn't figure out where it was in the UI.  So assuming the were removed, this now makes sense.

**Comment by okainov** [comment]
From #1931

>One of the troubles with the old Environment was that it wasn't very useful 

Well, we still don't agree as it was the killer-feature of Kiwi why we even chose it in our team. And we still (4 years after) use it incredibly actively.

>A new proposal is to choose the environment group and have test runs/test executions automatically created for you

This doesn't really help as what is needed is still possibility to add/modify environment/properties of a TR. Generation of multiple TRs by some matrix is IMO orthogonal to Environment.


I've just checked on demo page and I don't even see how "properties" can be used to achieve the same goal as Environment was used for. I see it's only available for Test Case and not for a TR.

So I don't see as this current issue #1334 can be closed, it's still open, nothing was done in regards of it unfortunately(?)

**Comment by okainov** [comment]
We're now trying to rebase/upgrade our version of Kiwi, first when I saw "Environment" I got a bit excited, but after reading the [blog article](https://kiwitcms.org/blog/atodorov/2022/01/24/feature-showcase-test-matrix-generation/) it seems like I got completely misunderstood by @atodorov unfortunately :( What is even more frustrating, is seeing my screenshot being misinterpreted.

>Redistributable and Testing Type look like information-only attributes. They don't appear to have any relevance to the test matrix at all. The same information-only effect can be achieved both with environments and with tags.

That's true for **all** of the items from my screenshot. The way we use env is not to modify test matrix in any way. But to add more details about the product-under-test and make it possible to link multiple test runs together. Imagine the _suite_ which can be installed and installs `Driver A` with version `1.2.3` and `Driver B` with version `3.2.1`. And those drivers are tested in a different test plans. But we need a way to **link** the test run for the whole suite with the (many) test runs for its ingredients to have the whole testing picture.

When I use tags, it's still just a pure string. I would need to enter `driver a=1.2.3` and this doesn't really help me. Old environment actually had the list and not a free-text fields which is important.
